calico_node/allocate_ipip_addr/allocate_ipip_addr.go

```diff
@@ -0,0 +1,131 @@
+package allocateIPIPAddr
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// main sets the host's tunnel address to an IPIP-enabled address if
+// there are any available, otherwise sets the tunnel to nil.
+func main() {
+	cfg, err := client.LoadClientConfig("")
+	if err != nil {
+		panic(fmt.Sprintf("Error loading config: %s", err))
+	}
+	c, err := client.New(*cfg)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating client: %s", err))
+	}
+	nodename := os.Getenv("NODENAME")
+
+	meta := api.IPPoolMetadata{}
+	ipPoolList, err := c.IPPools().List(meta)
+	if err != nil {
+		log.Warnf("error retrieving pools: %s", err)
+	}
+	fmt.Println(ipPoolList)
+	enabledPools := getIPIPEnabledPools(ipPoolList.Items)
```

Comment:
`getIPIPEnabledPools()` is really more of a "filter" than a "get".  I would move lines 27-32 into the getIPIPEnabledPools() method so that all the pools enumeration is encapsulated together,